sasmodels/direct_model.py

```diff
@@ -424,6 +426,12 @@ def test_reparameterize():
     derived_Iq = derived_calculator(volume=4*pi/3*Rp*Re**2, eccentricity=Rp/Re)
     assert norm((base_Iq - derived_Iq)/base_Iq) < 1e-13
 
+    # clean up the dll that was created in the cache as part of the test
+    try:
+        os.remove(derived_model.dllpath)
+    except Exception:
+        pass
```

Comment:
Thinking about why you need the try/except I realize it's because you may not have a dll path associated with the kernel, if it is a cuda kernel for example. Which begs the question of why dllpath is part of the "public" interface. We should probably rename it to _dllpath in the kerneldll.py so that you can't do this operation.

Instead we should have a `clean` method as part of the kernel API for removing the cache for one-offs like this, and this method can reference _dllpath.

Feel free to move this action to a new "clean" method, or merge this as is and add a ticket to address the technical debt.